lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -9,10 +9,223 @@
+                return instance['uuid']
+        return None
+
+    def get_by_uuid(self, uuid):
+        """ Returns an application as a dict specified by UUID, None otherwise
+        """
+        api_response = self.api.get_application_using_get(uuid)
+        if api_response:
+            return api_response.to_dict()
+        return None
+
+    def create(self, body):
+        """ Creates an application with the provided API request body, returns
+            bool of success/failure
+        """
+        api_response = self.api.create_application_from_template_using_post(
+            body)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return api_response.object_uuid
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def delete(self, uuid):
+        """ Deletes an application specified by UUID, returns bool of success/failure
+        """
+        api_response = self.api.delete_application_using_delete(uuid)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return True
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def power_action_on_instance_by_uuid(self, uuid, power_action):
+        """ Performs a specified power operation on an application instance
+            specified by UUID, returns bool of success/failure
+        """
+        api_response = power_action_dict[power_action](uuid)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return True
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+
+class VlanResource(object):
+    def __init__(self, api_client):
+        self.api = tacp.VlansApi(api_client)
+        self.api_client = api_client
+
+    def get_all(self):
+        """ Get all VLAN networks as a list of dicts
+        """
+        api_response = self.api.get_vlans_using_get()
+        vlan_dicts = []
+        if api_response:
+            for vlan in api_response:
+                vlan_dicts.append(vlan.to_dict())
+        return vlan_dicts
+
+    def get_uuid_by_name(self, name):
```

Comment:
Actually, now that I see what you are doing here, all of these methods are doing the same thing. `get_all`, `get_uuid_by_name`, `create`, `delete`, you can just implement them in your base class and the separate resource classes that inherit only have to define the `model` class.